kolibri/core/assets/src/components_demo.js

```diff
@@ -109,6 +110,173 @@ var TextInputWithTagDisplay = Mn.LayoutView.extend({
+            });
+            return match === undefined;
+        }, this));
+
+        // Used as template context variables
+        var userModelToHash = function(user) {
+            var attrs = _.clone(user.attributes);
+            _.extend(attrs, {
+                cid: user.cid
+            });
+            return attrs;
+        };
+        this.usersHash = _.map(this.users, userModelToHash);
+        this.usersToAddHash = _.map(this.usersToAdd, userModelToHash);
+
+        _.bindAll(this, 'template');
+    },
+
+    events: {
+        'click .delete': 'onClickDelete',
+        'click .ko_drop_content a': 'onAddUser'
+    },
+
+    onClickDelete: function(ev) {
+        var cid = $(ev.target).data('cid');
+        var user = _.find(this.users, function(user){
+            return user.cid === cid;
+        });
+        var excludeName = this.model.get('name');
+        user.set('classrooms', _.filter(user.get('classrooms'), function(name){
+            return name !== excludeName;
+        }));
+    },
+
+    onAddUser: function(ev) {
+        var cid = $(ev.target).data('cid');
+        var user = _.find(this.usersToAdd, function(user){
+            return user.cid === cid;
+        });
+        var crName = this.model.get('name');
+        var newCrs = _.clone(user.get('classrooms'));
+        newCrs.push(crName);
+        user.set('classrooms', newCrs);
+    }
+});
+
+
+var ClassroomCollection = Mn.CollectionView.extend({
+    childView: ClassroomView,
+
+    tagName: 'ul',
+
+    className: 'ko_list'
+});
+
+
+// This is not sufficiently general to be a first-class component.
+// It's just a thin wrapper around other components anyway.
+var ClassRosterView = Mn.LayoutView.extend({
+    template: _.template('<div class="heading">Class Roster</div>' +
+                         '<div class="classListRegion"></div>'),
+
+    regions: {
+        classList: '.classListRegion'
+    },
+
+    initialize: function() {
+        var classrooms = this.model.get('classrooms');
+        var users = this.model.get('users');
+        this.classList = new ClassroomCollection({
+            collection: classrooms,
+            childViewOptions: { // childViewOptions are passed to the initialize function of each child view
+                users: users
+            }
+        });
+    },
+
+    onBeforeShow: function() {
+        this.showChildView('classList', this.classList);
+    }
+});
+
+
+// This is the main view of the User Management demo.
+var UserManagementView = Mn.LayoutView.extend({
+    template: _.template('<div class="userListContainer">' +
+                            '<div class="heading">Users</div>' +
+                            '<div class="userListRegion"></div>' +
+                         '</div>' +
+                         '<div class="classRosterRegion"></div>'),
+
+    regions: {
+        userList: '.userListRegion',
+        classRoster: '.classRosterRegion'
```

Comment:
Here it looks like the class `classRosterRegion` is being used for two purposes - styling and child view insertion. I think this double-use is potentially risky because it couples two concepts that should be independent, making it more likely to introduce bugs if one aspect is refactored.